### PR TITLE
runtime: Avoid some gcc specific errors

### PR DESCRIPTION
**Pull request by robryanx**
For better support with building with gcc as part of a serenity build

### CODE REVIEW

File: runtime/Jakt/Platform.h [hunk 1]
```diff
@@ -42,7 +42,7 @@
 #ifdef ALWAYS_INLINE
 #    undef ALWAYS_INLINE
 #endif
-#define ALWAYS_INLINE
+#define ALWAYS_INLINE __attribute__((always_inline)) inline
```
Comment (ADKaster) [inline runtime/Jakt/Platform.h:45]: This was intentionally removed I think. Why is it needed?

File: runtime/Jakt/String.h [hunk 2]
```diff
@@ -19,6 +19,11 @@
 
 namespace Jakt {
 
+bool operator<(char const* characters, String const& string);
```
Comment (ADKaster) [inline runtime/Jakt/String.h:22]: What code complains when these aren't here? Where are they implemented?

File: runtime/Jakt/StringBuilder.h [hunk 3]
```diff
@@ -49,7 +49,7 @@ class StringBuilder {
     [[nodiscard]] bool is_empty() const { return m_buffer.is_empty(); }
 
     template<class SeparatorType, class CollectionType>
-    ErrorOr<void> join(SeparatorType const& separator, CollectionType const& collection, StringView fmtstr = "{}"sv)
+    void join(SeparatorType const& separator, CollectionType const& collection, StringView fmtstr = "{}"sv)
```
Comment (ADKaster) [inline runtime/Jakt/StringBuilder.h:52]: This looks wrong. Now ErrorOrs from append and appendff will be dropped on the floor -- which shouldn't be possible, ErrorOr is [[nodiscard]]

File: runtime/Jakt/Platform.h (hunk 1 above)
Comment (robryanx) [reply runtime/Jakt/Platform.h:45]: This error is from:

runtime/Jakt/Memory.h:19:20: error: no previous declaration for 'void fast_u32_copy(u32*, const u32*, size_t)' [-Werror=missing-declarations]
ALWAYS_INLINE void fast_u32_copy(u32* dest, u32 const* src, size_t count)

runtime/Jakt/Memory.h:30:20: error: no previous declaration for 'void fast_u32_fill(u32*, u32, size_t)' [-Werror=missing-declarations]
ALWAYS_INLINE void fast_u32_fill(u32* dest, u32 value, size_t count)

File: runtime/Jakt/String.h (hunk 2 above)
Comment (robryanx) [reply runtime/Jakt/String.h:22]: This error for each of them:

runtime/Jakt/String.cpp:174:6: error: no previous declaration for 'bool Jakt::operator<(const char*, const String&)' [-Werror=missing-declarations]

File: runtime/Jakt/StringBuilder.h (hunk 3 above)
Comment (robryanx) [reply runtime/Jakt/StringBuilder.h:52]: Yeah I think this is the wrong fix looking now, it probably just needs a return at the end, I think `return {};`

**Comment by robryanx** [comment]
Closing this, as of today seems to compile fine with gcc and the existing options.

**Pull request closed without merging**